mlx/backend/cuda/metal.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright © 2023-2024 Apple Inc.
+
+#include "mlx/backend/metal/metal.h"
+#include "mlx/backend/cuda/device.h"
+#include "mlx/backend/metal/metal_impl.h"
+#include "mlx/primitives.h"
+#include "mlx/scheduler.h"
+
+#include <nvtx3/nvtx3.hpp>
+
+namespace mlx::core::metal {
```

Comment:
I think it's worth generalizing this part of the code. 

For example, checking `metal::is_available` doesn't really make sense for the CUDA back-end.

Here's some options:

- Add a `mx::is_available(Device)` which checks on the given device. 
- For the GPU device we have a shared header that lives in `backend/gpu` and in the namespace `gpu`. 
- The implementations then go in `metal`, `cuda`,  and `no_gpu` respectively.
- We could potentially mark `metal::is_available` as deprecated as well.. 

In general a good aim is to work towards having device-general code. So in user-land it would be really nice to get rid of all those calls to `metal.is_available()` and instead check for the device.  And if we need back-end specific availability checks have them as internal API.

Reply:
Another option is to put these in the `mx_cuda` namespace and duplicate some of the headers. I think this is suboptimal though because then we need to duplicate stuff in calling code.